### PR TITLE
[MM-15381] Write an automated test using Cypress for "Whitespace with emojis does not affect size"

### PR DESCRIPTION
**Pull request by Jonany**
#### Summary
Adds test for whitespace's effect on emojis.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10777

### CODE REVIEW

File: cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    // Unpin all posts at the end of the test
+    after(() => {
+        // pinnedPosts.forEach((pinnedPost) => {
+        //     cy.apiUnpinPosts(pinnedPost);
+        // });
+    });
```
Comment (saturninoabril) [inline cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js]: Maybe missed to remove after copy/paste?

File: cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (saturninoabril) [inline cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js]: We'll re-organize the folder structure of integration soon.  With that, could you please create new folder and rename this file into something like:
```
cypress/integration/messaging/emoji_size_spec.js 
```

**Comment by saturninoabril** [comment]
One more thing: build is failing due to outdated branch.  Could you please update by merging master into this branch?

**Comment by Jonany** [comment]
@saturninoabril @thekiiingbob Thanks for the feedback! I'm planning on working on the changes you mentioned tomorrow.

File: cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js (hunk 1 above)
Comment (Jonany) [reply cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js]: Removed

File: cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js (hunk 2 above)
Comment (Jonany) [reply cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js]: Done!

File: cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+        // * Verify message is visible and does not have a new line
+        cy.get('@newLineMessage').
+            should('be.visible').
+            and('not.contain', '\n');
+
+        // * Verify book emoticon size
+        cy.get('@newLineMessage').
+            find('span[alt=":book:"]').
+            and('have.css', 'minHeight', '32px').
```
[the remark this answers is not in the document]
Comment (Jonany) [reply cypress/integration/onboarding/messaging/whitespace_does_not_affect_emoji_size_spec.js]: Thanks for the suggestion! Moved check code into a function and consolidated all the emoji name references down to one constant.

File: cypress/integration/messaging/emoji_size_spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        // * Verify message is visible and starts with three spaces
+        cy.get('@spacesMessage').
+            should('be.visible');
+        cy.get('@spacesMessage').
+            should((message) => {
+                expect(message.find('span.all-emoji p').html()).to.match(/^[ ]{3}/);
+            });
+
+        // * Verify emoji size
+        checkEmojiSize('@newLineMessage', emojis);
```
Comment (thekiiingbob) [inline cypress/integration/messaging/emoji_size_spec.js]: This should be checking `@spacesMessage` right?

File: cypress/integration/messaging/emoji_size_spec.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/*eslint max-nested-callbacks: ["error", 5]*/
+
+function checkEmojiSize(message, emojis) {
+    emojis.forEach((emoji) => {
+        cy.get('@newLineMessage').
```
Comment (thekiiingbob) [inline cypress/integration/messaging/emoji_size_spec.js]: I think you meant to use the `message` variable where you are using `@newLineMessage`.

File: cypress/integration/messaging/emoji_size_spec.js (hunk 4 above)
Comment (Jonany) [reply cypress/integration/messaging/emoji_size_spec.js]: Yes. Thanks for catching that.

File: cypress/integration/messaging/emoji_size_spec.js (hunk 5 above)
Comment (Jonany) [reply cypress/integration/messaging/emoji_size_spec.js]: Yes. Thanks for catching that.

**Pull request merged**